website/docs/getting-started/basics.md

```diff
@@ -8,15 +8,34 @@ odo abstracts Kubernetes concepts into developer focussed terminology. In this d
 
 ### Concepts
 
-#### Component
+* Component
+  A component is a microservice. Each component can be run and deployed separately. Examples of components: Warehouse API Backend, Inventory API, Web Frontend, Payment Backend
+  > A component in odo is an equivalent of a deployment or deploymentconfig in Kubernetes. Creating a new component in odo will create a deployment in Kubernetes or for git and binary based component creation, it will create deploymentconfig.
 * Application
+  An application consists of multiple components which may span over multiple projects, and work individually to build the entire application. Examples of applications: e-Shop, Hotel Reservation System, Online Booking
+  > An application can be considered as an equivalent of labels in Kubernetes that help in grouping a set of resources.
 * Project
+  A project is your source code, tests, and libraries organized in a separate single unit.
+  > A project in odo is an equivalent of a namespace in Kubernetes. Creating a project in odo will create a namespace in Kubernetes with the same name.
 * Context
+  A context is the directory where the source code, tests, and libraries for your component resides. A single context can only contain a single component.
 * URL
+  A URL exposes your component to be accessed outside the cluster.
+  > A URL is an equivalent of a service in Kubernetes.
```

Comment:
NO! It's not an equivalent of a Service in k8s! It's equivalent to Ingresses in k8s and Routes in OCP.

Reply:
:grimacing: Okay. Changing that.

Reply:
The OCP part was just for my fyi, right? Do we want to mention it in the docs?

Reply:
Yeah. Even the Ingress thing was for reference. We need to explain URL without talking about Ingresses or Routes.